### DISCUSSION TITLE
Companion object with return type

### DISCUSSION THREAD

**Discussion by cedmart1decath** [square/kotlinpoet#1678] [Q&A]
**Is your feature request related to a problem? Please describe.**
I am trying to generate a singleton class using companion object

**Describe the solution you'd like**
I'd like to be able to output the correct companion object definition

**Describe alternatives you've considered**
None atm
**Additional context**
Example 
```
class MySingletonClass {

  fun doSomething() {
      println("Hello world")
  }

   companion object : SingletonHolder<MySingletonClass>(::MySingletonClass)
}
```
Singleton Holder class => https://bladecoder.medium.com/kotlin-singletons-with-argument-194ef06edd9e

**Comment by Egorand** [comment]
Not clear what you've tried so far and what didn't work for you.

**Comment by cedmart1decath** [comment]
So far I have tried to use TypeSpec.companionObjectBuilder() but the building chain does not provide a way to specify a type

**Comment by cedmart1decath** [comment]
Nevermind I was not headed in the right direction. I'll use TypeSpec.objectBuilder instead
